firebase-firestore/src/main/java/com/google/firebase/firestore/Query.java

```diff
@@ -84,6 +84,77 @@ public FirebaseFirestore getFirestore() {
     return firestore;
   }
 
+  /** Validates the given document ID and returns the corresponding FieldValue. */
+  private FieldValue validateAndCalculateReferenceValue(FieldPath fieldPath, Object value) {
```

Comment:
We usually organize our code "top down" where helper methods live below the point where they're used...  so I'd move these methods below whereHelper()...